testing/scenario_app/README.md

```diff
@@ -0,0 +1,42 @@
+# Scenario App
+
+This folder contains a dart:ui application and scripts to compile it to AOT
+for exercising embedders.
+
+It intentionally has no dependencies on the Flutter framework or tooling, such
```

Comment:
I'm curious what types of tests this is likely to facilitate. Do you already have some in mind? 

As I think out loud, we wouldn't be able to test with any plugins, right? 

We could verify that the embeddings send certain messages to Flutter, but we would be unable to verify any back-and-forth conversations that might occur, because the Flutter side would not really exist. Also, with something like dark mode, we can verify that the messages are sent to Flutter, but we can't truly do an e2e test that the dark mode feature, itself, is correctly implemented, because Flutter isn't there taking any actions.

What might this mean for the rendering system? For example, in the Android embedding, we have FlutterView, FlutterSurfaceView, FlutterTextureView, FlutterRenderer, FlutterRenderSurface...would all of these be excluded from significant testing in the absence of Flutter framework rendering? Or do you think there are testing strategies that effectively puts them through their paces without building out significant infrastructure?

Android now has splash capabilities, which will be used for most/all full-Flutter apps, but that functionality depends upon a message from Flutter letting us know that the 1st frame has been rendered. Without that message, the splash will never disappear. Is that going to be an issue with this testing approach?

Reply:
We can send whatever messages we want to the embedding from the Dart code.  We  could write scenarios up that mimic the way the framework or a plugin would send messages.

Testing plugins with this setup would be a chore but might be possible.

I'm not sure how to answer the rendering system question - the current demo renders a moving color changing square. We could create more or less complicated things to render. We would not want to invest lots of time making those things look like Material widgets, and testing that's concerned about that probably belongs in the framework.  but we certainly could create an Android app that uses various embedding classes to do interesting things (and write tests for them in the app) - right now, the Android app is just a barebones, dead simple implementation that mimics mostly what the vanilla Flutter app does.